controllers/registry.go

```diff
@@ -262,8 +262,11 @@ func (r *BackOffRegistry) Get(objKey client.ObjectKey) (*BackOff, bool) {
 
 	entry, ok := r.m[objKey]
 	if !ok {
+		bo := backoff.NewExponentialBackOff(r.opts...)
+		// call Reset() to ensure that the initial interval is honoured.
+		bo.Reset()
```

Comment:
Do we still need the wrapper for backoff.Backoff on line 279? The comment there says it purposefully doesn't implement Backoff.Reset.

Reply:
I had considered it. I am okay to leave it as-is since it may still prove useful if we ever want to add instrumentation.